### PR TITLE
F lca swift tests

### PR DESCRIPTION
**Pull request by janniclas**
Besides many unit tests for LCA for Swift-based LLVM IR, a utility function to check llvm types for "integer like"  structs (isIntegerLikeType) was added. Further, the LCA's getNormalEdge function was fixed to correctly take the getElementPtr workaround from the getNormalFlow function into account and correctly generate the edge for the desired alloca instruction.

### CODE REVIEW

**Comment by janniclas** [comment]
Global Variables follow this pattern in Swift-based LLVM IR.

```
@"$s9global_012g1Sivp" = hidden global %TSi <{ i64 10 }>, align 8, !dbg !0

define hidden swiftcc i8* @"$s9global_012g1Sivau"() #0 !dbg !41 {
entry:
  ret i8* bitcast (%TSi* @"$s9global_012g1Sivp" to i8*), !dbg !45
}
```

Currently, we don't support this behavior, however all other test cases translated from C++ work as expected with the changes included in this PR. 
I'll extend the LCA with support for globals at a later point

File: lib/PhasarLLVM/DataFlow/IfdsIde/Problems/IDELinearConstantAnalysis.cpp [hunk 1]
```diff
@@ -287,10 +287,14 @@ IDELinearConstantAnalysis::~IDELinearConstantAnalysis() {
 IDELinearConstantAnalysis::FlowFunctionPtrType
 IDELinearConstantAnalysis::getNormalFlowFunction(n_t Curr, n_t /*Succ*/) {
   if (const auto *Alloca = llvm::dyn_cast<llvm::AllocaInst>(Curr)) {
-    if (Alloca->getAllocatedType()->isIntegerTy()) {
+    auto *AT = Alloca->getAllocatedType();
+    if (AT->isIntegerTy()) {
+      return generateFromZero(Alloca);
+    } else if (isIntegerLikeType(AT)) {
       return generateFromZero(Alloca);
     }
```
Comment (fabianbs96) [inline lib/PhasarLLVM/DataFlow/IfdsIde/Problems/IDELinearConstantAnalysis.cpp:295]:
```suggestion
    if (AT->isIntegerTy() || isIntegerLikeType(AT)) {
      return generateFromZero(Alloca);
    }
```

File: lib/PhasarLLVM/Utils/LLVMShorthands.cpp [hunk 2]
```diff
@@ -64,6 +64,14 @@ bool isFunctionPointer(const llvm::Value *V) noexcept {
   return false;
 }
 
+bool isIntegerLikeType(const llvm::Type *T) noexcept {
+  if (const auto *structType = llvm::dyn_cast<llvm::StructType>(T)) {
+    return structType->isPacked() && structType->elements().size() == 1 &&
+           structType->getElementType(0)->isIntegerTy();
+  }
```
Comment (fabianbs96) [inline lib/PhasarLLVM/Utils/LLVMShorthands.cpp:71]:
```suggestion
  if (const auto *StructType = llvm::dyn_cast<llvm::StructType>(T)) {
    return StructType->isPacked() && StructType->elements().size() == 1 &&
           StructType->getElementType(0)->isIntegerTy();
  }
```

File: unittests/PhasarLLVM/DataFlow/IfdsIde/Problems/IDELinearConstantAnalysisSwiftTest.cpp [hunk 3]
```diff
@@ -20,32 +22,32 @@ class IDELinearConstantAnalysisSwiftTest : public ::testing::Test {
...
       "linear_constant/"); // Function - Line Nr - Variable - Value
+  const std::vector<std::string> EntryPoints = {"main"};
+
+  // Function - Line Nr - Variable - Value
   using LCACompactResult_t = std::tuple<std::string, std::size_t, std::string,
                                         IDELinearConstantAnalysisDomain::l_t>;
-  std::unique_ptr<LLVMProjectIRDB> IRDB;
 
-  void SetUp() override {}
+  void SetUp() override { ValueAnnotationPass::resetValueID(); }
```
Comment (fabianbs96) [inline unittests/PhasarLLVM/DataFlow/IfdsIde/Problems/IDELinearConstantAnalysisSwiftTest.cpp:31]:
```suggestion
```

This is not needed anymore. The LLVMProjectIRDB's ctors do not use the ValueAnnotationPass anymore

File: unittests/PhasarLLVM/DataFlow/IfdsIde/Problems/IDELinearConstantAnalysisSwiftTest.cpp [hunk 4]
```diff
@@ -81,17 +83,583 @@ class IDELinearConstantAnalysisSwiftTest : public ::testing::Test {
...
+  auto Results = doAnalysis("while_04.ll");
+  std::set<LCACompactResult_t> GroundTruth;
+  GroundTruth.emplace("$s8while_046MyMainV4mainyyFZ", 4, "i", 42);
+  GroundTruth.emplace("$s8while_046MyMainV4mainyyFZ", 7, "a", 0);
+  compareResults(Results, GroundTruth);
+}
+
+/* ============== Global TESTS ============== */
+
+// TEST_F(IDELinearConstantAnalysisSwiftTest, HandleGlobalsTest_01) {
```
Comment (fabianbs96) [inline unittests/PhasarLLVM/DataFlow/IfdsIde/Problems/IDELinearConstantAnalysisSwiftTest.cpp:426]: You may want to enable the commented code below and mark the tests as disabled by starting their name with `DISABLED_` or by calling the macro `GTEST_SKIP()` from within the test, e.g. 
```C++
GTEST_SKIP() << "Swift-globals are not supported yet";
```

**Review by fabianbs96 — APPROVED** [review]
lgtm except minor

**Pull request merged**